linux_os/guide/system/permissions/restrictions/sysctl_kernel_perf_event_paranoid/rule.yml

```diff
@@ -0,0 +1,23 @@
+documentation_complete: true
+
+prodtype: rhel8,fedora,ol8
+
+title: 'Disallow kernel profiling by unprivileged users'
+
+description: '{{{ describe_sysctl_option_value(sysctl="kernel.perf_event_paranoid", value="2") }}}'
+
+rationale: |-
+    Kernel profiling can reveal sensitive information about kernel behaviour.
+
+severity: medium
+
+identifiers:
+    cce@rhel8: 81054-9
+
+references:
+    anssi: NT28(R25)
+    ospp: FMT_SMF_EXT.1
+
+{{{ complete_ocil_entry_sysctl_option_value(sysctl="kernel.perf_event_paranoid", value="2") }}}
+
+platform: machine
```

Comment:
Is this truly only applicable to bare metal and virtual machines?

Reply:
What other types of machines are there, for the `platform` value?

Reply:
I think @shawndwells meant applicability to containers. Because even though you are in a container environment, you may have the kernel package installed even though it is not going to be run. Our rules which touch these kernel configuration are already restricting the applicability to `machine`, e.g. https://github.com/ComplianceAsCode/content/blob/af36c2a29e5747f287b44922825c96a2a3992e17/linux_os/guide/system/permissions/restrictions/sysctl_kernel_kexec_load_disabled/rule.yml#L20

If we decide to change the applicability we need to consider other rules as well.

Reply:
I think this should be kept as `machine` only.
I could not find mention that `kernel.perf_event_paranoid` is safe or namespaced.

Reply:
To elaborate, yes, @ggbecker was right. I was asking if this was applicable inside containers. Checking the kernel.org and red hat documentation there seems to be very limited docs on what sysctls are applicable inside containers and which ones are not.